### ISSUE TITLE
v4: `Col` API discussion 

### ISSUE THREAD

**Issue by jquense** [react-bootstrap/react-bootstrap#3103]
I wanted to quickly layout and discuss some options for the `Col` component api as we add v4 support. There are only a few things here, the `col-*` spans, `order-*` (flex order), and `offset-*` classes, each with a breakpoint option as well (`col-8`, `col-lg-8`, etc)

potential options:

### The current approach

```
<Col md={true} xl={3} mdOffset={3} mdOrder={1}> // .col-md, .col-xl-3, .offset-md-3 .order-md-1
```
**pros:** Nice and flat
**cons:** a bit hard to parse overall

### Breakpoint first

```
<Col xl={3} md={{ span: true, offset: 3, order: 1 }}>
```
**pros:** clearly organized breakpoint configuration
**cons:** maybe the simple non responsive case is to complicated? e.g. `xs={{ span: 1, offset: 1 }}`

### ClassName first

```
<Col span={{ xs: true, lg: 3 }} order={1} offset={{ sm: 1, lg: 3 }}>
```
**pros:** The simple case (more common?) case is easy:`span order={1} offset={2}`
**cons:** the responsive case is harder to understand

We should also think of this in light of #3089 and the API choices going on there for consistency.

**Comment by taion** [comment]
This will break pure-render style optimizations, though, no?

**Comment by jquense** [comment]
Yes, but we aren't doing pure render here anyway? It's easy enough to hoist the prop values tho

**Comment by Shawnxkwang** [comment]
I like the Breakpoint first approach, more intuitive to me at least.
